TestRaceMissing fails on non-amd64 platforms

https://buildd.debian.org/status/logs.php?pkg=gb&ver=0.4.0-1 shows what I mean (https://buildd.debian.org/status/fetch.php?pkg=gb&arch=i386&ver=0.4.0-1&stamp=1453837841) -- if the race detector isn't supported on a platform, `TestRaceMissing` fails fatally:
```
=== RUN   TestRaceMissing
--- FAIL: TestRaceMissing (0.02s)
	gb_test.go:179: running /tmp/testgb130146059/testgb [test -race]
	gb_test.go:193: standard error:
	gb_test.go:194: FATAL: race detector not supported on linux/386
		
	gb_test.go:213: /tmp/testgb130146059/testgb failed as expected: exit status 1
	gb_test.go:255: expected missing race support message
	gb_test.go:256: pattern FATAL: go installation at /usr/lib/go is missing race support not found in standard error
```

Any way we could make this skip easily without me just skipping the tests on all non-amd64 platforms (or patching it out, which is kinda gross)? :smile:

--- a/cmd/gb/gb_test.go
+++ b/cmd/gb/gb_test.go
@@ -1354,7 +1354,9 @@ func TestRaceMapRW(t *testing.T) {
 	tmpdir := gb.tempDir("tmp")
 	gb.setenv("TMP", tmpdir)
 	gb.runFail("test", "-race")
-	gb.grepStderr(regexp.QuoteMeta(fmt.Sprintf("FATAL: go installation at %s is missing race support", runtime.GOROOT())), "expected missing race support message")
+	raceError1 := fmt.Sprintf("FATAL: go installation at %s is missing race support", runtime.GOROOT())
+	raceError2 := fmt.Sprintf("FATAL: race detector not supported on %s/%s", runtime.GOOS, runtime.GOARCH)
+	gb.grepStderr(regexp.QuoteMeta(raceError1)+"|"+regexp.QuoteMeta(raceError2), "expected missing race support message")
 	gb.mustBeEmpty(tmpdir)
 }
 

